src/storage/disk_manager.c

```diff
@@ -3855,15 +3855,17 @@ disk_rv_reserve_sectors (THREAD_ENTRY * thread_p, LOG_RCV * rcv)
       /* disk check is in progress. unfix page, get critical section again and then fix page and make the changes */
       VPID vpid;
       pgbuf_get_vpid (rcv->pgptr, &vpid);
-      pgbuf_unfix_and_init (thread_p, rcv->pgptr);
+      pgbuf_unfix (thread_p, rcv->pgptr);
+      memset (rcv->pgptr, 0, sizeof (char *));
```

Comment:
- Please check pgbug_unfix_and_init again. It doesn't reset the data in page, it only resets the page pointer as NULL.
- What this function does, unfixing & fixing page again, is quite a tricky work-around. But we have no choice at this time but to keep it. Add a comment and strip the const.
```suggestion
      // to avoid a deadlock between CSECT_DISK_CHECK and page fix, we are forced to unfix the page here.
      // make an exception to the const rule.
      // *INDENT-OFF*
      pgbuf_unfix (thread_p, (const_cast<LOG_RCV *> (rcv))->pgptr);
      // *INDENT-ON*
```